
libpthread isn't needed because of the TIB work.  libutil, well, I don't 
think it's been needed for a long time, if ever; just a generic configure 
script in action.

I've been running my laptop with this for a couple days.

Amusingly, this is mostly polish: the new libpthread doesn't install its 
callbacks until a process does something that requires the threading bits 
to initialize, so this just saves a bit of memory and a bit of ld.so's 
time.


Philip

Index: hints/openbsd.sh
===================================================================
RCS file: /cvs/src/gnu/usr.bin/perl/hints/openbsd.sh,v
--- ./hints/openbsd.sh.orig	Sat Dec 24 14:58:07 2022
+++ ./hints/openbsd.sh	Sat Dec 24 14:58:09 2022
@@ -96,6 +96,9 @@ esac
 # around for old NetBSD binaries.
 libswanted=`echo $libswanted | sed 's/ crypt / /'`
 
+# OpenBSD hasn't ever needed linking to libutil
+libswanted=`echo $libswanted | sed 's/ util / /'`
+
 # Configure can't figure this out non-interactively
 d_suidsafe=$define
 
@@ -153,6 +156,9 @@ $define|true|[yY]*)
         	# Broken up to OpenBSD 3.6, fixed in OpenBSD 3.7
 		d_getservbyname_r=$undef ;;
 	esac
+	;;
+*)
+	libswanted=`echo $libswanted | sed 's/ pthread / /'`
 esac
 EOCBU
 
